app/extensions/email/__init__.py

```diff
@@ -184,6 +167,32 @@ def template(self, template, **kwargs):
             pmail = Premailer(**pmail_kwargs)  # REF: https://pypi.org/project/premailer/
         log.info('Using premailer = %r' % (pmail))
 
+        from app.modules.site_settings.models import SiteSetting
+
+        # Default values for templates
+        kwargs.setdefault(
+            'header_image_url', SiteSetting.get_value('email_header_image_url')
+        )
+        kwargs.setdefault('h1', SiteSetting.get_value('email_title_greeting'))
+        kwargs.setdefault(
+            'secondary_title', SiteSetting.get_value('email_secondary_title')
+        )
+        kwargs.setdefault('secondary_text', SiteSetting.get_value('email_secondary_text'))
+        kwargs.setdefault(
+            'legal_statement', SiteSetting.get_value('email_legal_statement')
+        )
+        kwargs.setdefault('unsubscribe_link', '/unsubscribe')
+        kwargs.setdefault(
+            'site_name', SiteSetting.get_value('site.name', default='Codex')
+        )
+        kwargs.setdefault('site_url', '/')
+        kwargs.setdefault('site_domain', 'flukebook.org')
```

Comment:
That looks wrong.

Reply:
Let me see where we have the domain setting...

Reply:
```diff
@@ -186,7 +196,10 @@ class Email(Message):
             'site_name', SiteSetting.get_value('site.name', default='Codex')
         )
         kwargs.setdefault('site_url', '/')
-        kwargs.setdefault('site_domain', 'flukebook.org')
+        kwargs.setdefault(
+            'site_domain',
+            urlparse(current_app.config.get('BASE_URL')).netloc.split(':')[0],
+        )
         kwargs.setdefault('twitter_url', SiteSetting.get_value('twitter_url'))
         kwargs.setdefault('facebook_url', SiteSetting.get_value('facebook_url'))
         kwargs.setdefault(
```

Reply:
this seems to be a fussy thing to figure out.  i had tried to basically come up with code to get "the site url" here:
https://github.com/WildMeOrg/houston/blob/main/app/utils.py#L42-L53

i am not sure the best way to make this work in both a production and dev environment.    i am curious to see if your code above does the expected thing on, say, the QA server (which managed to not work at all with my code).

Reply:
Thanks, I'm using `site_url_prefix()` instead of `BASE_URL`.  I changed `site_url_prefix()` to return `localhost:84` instead so it's the same as the development set up